### ISSUE TITLE
Caliburn.Micro.Platform.Core should have its own package

### ISSUE THREAD

**Issue by godrose** [Caliburn-Micro/Caliburn.Micro#306]
The type ``AssemblySource`` is defined in the ``Caliburn.Micro.Platform.Core`` project which is portable.
However I'm unable to add it from nuget to a PCL as this assembly is part of **Caliburn.Micro** package which has no portable target.
I can cheat and add the assembly explicitly. It works. Yet in my opinion this should be resolved at the packaging level.

**Comment by nigel-sampson** [comment]
Can I ask what you're doing with `AssemblySource` in a PCL?

`Caliburn.Micro.Platform.Core` ended up being refactored out of the main platform assembly because of Xamarin.Forms, while highly unlikely there could be two platforms running at the same time, the host one as well as Xamarin.Forms. The code required for both platforms couldn't be "file shared" between the projects without type clashes or duplicating the class in another namespace.

This assembly only contains a a few helper types that needed to be shared and wasn't intended to be used from a PCL.

I'd like to hear more about what you're looking to do.

**Comment by godrose** [comment]
Here is the setup:

I've been practicing TDD for more than two years. One of the neat features of this approach is having to write a test for each bug that is discovered in the application to ensure it won't return. Lots of times these bugs involved several steps and thus couldn't be reproduced in a unit test. So I started writing integration tests. These tests stretched from the View Model level to the Providers (sort of point of contact with the Server).
I simulated system initialization from the very start, including the bootstrapper.
The bootstrapper includes the stage of loading assemblies for the subsequent inspection and view model / view types loading. After the bootstrapper is executed once, the ``AssemblySource`` is populated. This class is **static** and therefore next time an integration test is run in the same session I get the ``DuplicateTypeException`` or something similar.
So the only way to work with ``Caliburn.Micro`` as the screen lifecycle provider and have the integration tests running is to call ``AssemlySource.Clear`` between the runs.
This does the job.
I've refactored this functionality to a separate nuget package which hides the implementation details and exposes only ``TestHelper`` class. 
I've been adding the UWP support to lots of my packages and converting them into PCL's where applicable. In the case of ``LogoFX.Testing.Shared.Caliburn.Micro`` package I only need the ``AssemblySource`` type which I believe should be defined inside a PCL.
You can see the package at https://github.com/LogoFX/logofx-client-testing-shared-caliburn-micro

**Comment by godrose** [comment]
@nigel-sampson Any updates?

**Comment by nigel-sampson** [comment]
Thanks for the information on how you're using it.

At the moment I probably won't create a new package for `Caliburn.Micro.Platform.Core`, there's a significant overhead in maintaining more packages as well having issues where people reference the wrong page and supporting that etc.

I also imagine the scenario you're hitting is reasonably uncommon, if this changes however I'd be willing to revisit it.

Working around this you can do a couple of things, first is obviously carrying on with referencing the assembly directly, another would be modify the app to selectively add to the bootstrapper like so:

``` csharp
protected override IEnumerable<Assembly> SelectAssemblies()
{
    var viewAssembly = typeof (MainViewView).Assembly;

    if (!AssemblySource.Instance.Contains(viewAssembly))
        yield return viewAssembly;
}
```

Which would remove the need for clearing `AssemblySource`.

**Comment by godrose** [comment]
Thanks for the response. I don't think the second option is really appropriate as it breaks the SRP with regard to the bootstrapper. The bootstrapper should be as decoupled from its usage context as possible yet by adding this check I would introduce "awareness" to the fact that it will be used in tests, which is not always true. One may say that this is more like Defensive Programming stuff and this might be correct too.
To sum up, I'll keep referencing the assembly.
Thanks for your time.
